src/near/schema.ts

```diff
@@ -4,42 +4,64 @@ import { readCustomSection } from "wasm-walrus-tools"
 import { ContractCodeView } from "near-api-js/lib/providers/provider"
 import { JSONSchema7 } from "json-schema"
 
+
 export async function fetchSchema(contract: string, near: naj.Near): Promise<JSONSchema7> {
   // TODO handle either HTTP endpoint or IPFS hash
-  const url = await fetchJsonAddress(contract, near)
+  const url_or_data = await fetchJsonAddressOrData(contract, near)
+
 
   // TODO cache schema JSON in localeStorage, return early here if available
 
-  const schema = fetch(url)
+  if (url_or_data.startsWith("https://")) {
+    const schema = fetch(url_or_data)
     .then((response) => {
       if (!response.ok) throw new Error(`HTTP error! Status: ${response.status}`)
       return response.json()
     })
+    return schema;
+  }
 
   // TODO validate schema adheres to JSONSchema7
-  return schema
+  return JSON.parse(url_or_data)
 }
 
-class NoCustomSectionError extends Error {
+class NoCustomSection extends Error {
```

Comment:
rude!

Reply:
(sort of joking; I don't really care what we call this variable. But I do find the rename to be a pointless change that adds noise to this PR)

Reply:
Fair enough, just a pet peeve of mine.  `throw new NoCustomSection` is clearly an error.  Can remove if you find it too noisy.

Reply:
I'll let it slide 🙃